project_task_materials_stock/models/project.py

```diff
@@ -73,7 +73,14 @@ def write(self, vals):
                         task.material_ids.create_analytic_line()
                 else:
                     if task.unlink_stock_move():
-                        task.analytic_line_ids.unlink()
+                        if any(task.material_ids.mapped(
+                                'analytic_line_id.invoice_id')):
+                            raise exceptions.Warning(
+                                _(
+                                    "You can't move to not consume stage "
```

Comment:
You can't move to a not consume stage if there are already invoiced analytic lines